src/lib/io/sol-gpio-impl-zephyr.c

```diff
@@ -179,32 +198,24 @@ sol_gpio_open_raw(uint32_t pin, const struct sol_gpio_config *config)
 SOL_API void
 sol_gpio_close(struct sol_gpio *gpio)
 {
-    struct sol_gpio *g;
-    uint16_t i;
+    struct sol_gpio_port *port;
 
-    SOL_NULL_CHECK(device);
     SOL_NULL_CHECK(gpio);
 
-    SOL_PTR_VECTOR_FOREACH_IDX (&opened_pins, g, i) {
-        if (g->pin == gpio->pin) {
-            if (gpio_pin_disable_callback(device, g->pin) < 0) {
-                SOL_WRN("Couldn't disable gpio callback");
-            }
-            sol_ptr_vector_del(&opened_pins, i);
-            break;
-        }
-    }
+    port = gpio->port;
+    port->cb.pin_mask &= ~BIT(gpio->pin);
+    gpio_pin_disable_callback(port->device, gpio->pin);
```

Comment:
I guess this can cause problems for multiple `sol_gpio_open()`s, for example, two different pieces (A and B) of code call `_open()` for pin 10, when A calls `_close()` the callback is disabled for B also. Or am I missing something?